File: config/initializers/sentry.rb
Sentry.init do |config|
  config.enabled_environments = %w[staging production review]
  config.dsn = ENV["SENTRY_DNS"]
  config.breadcrumbs_logger = [:active_support_logger, :http_logger]

  if ENV['APP'] == 'mon-suivi-justice-staging'
    config.environment = 'staging'
  elsif ENV['APP'] == 'mon-suivi-justice-prod'
    config.environment = 'production'

  elsif ENV['APP']&.match?(/^mon-suivi-justice-staging-pr\d+/)
    config.environment = 'review'
  end

  # Set tracesSampleRate to 1.0 to capture 100%
  # of transactions for performance monitoring.
  # We recommend adjusting this value in production
  config.traces_sample_rate = 0.02
end

File: app/javascript/sentry.js
import * as Sentry from "@sentry/browser";

Sentry.init({
  dsn: SENTRY_DSN,
  environment: APP,
  release: 'msj@1.0.0',

  // Alternatively, use `process.env.npm_package_version` for a dynamic release version
  // if your build tool supports it.
  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],

  // Set tracesSampleRate to 1.0 to capture 100%
  // of transactions for performance monitoring.
  // We recommend adjusting this value in production
  tracesSampleRate: 1.0,

  // Capture Replay for 10% of all sessions,
  // plus for 100% of sessions with an error
  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 1.0,
});